backport_max_files_rewrite_option (#13082)
apache/iceberg: docs/docs/spark-procedures.md

#### Options

##### General Options

> | Name | Default Value                                                                                                                 | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |

Author: I did try to revert the documentation changes but for some reason git picks up changes which it isnt supposed to

| `min-input-files` | 5 | Any file group exceeding this number of files will be rewritten regardless of other criteria |
| `rewrite-all` | false | Force rewriting of all provided files overriding other options |
| `max-file-group-size-bytes` | 107374182400 (100GB) | Largest amount of data that should be rewritten in a single file group. The entire rewrite operation is broken down into pieces based on partitioning and within partitions based on size into file-groups.  This helps with breaking down the rewriting of very large partitions which may not be rewritable otherwise due to the resource constraints of the cluster. |

> | `max-files-to-rewrite` | null | This option sets an upper limit on the number of files eligible for rewrite operation. It can be useful for improving job stability, particularly when dealing with a large number of files. If this option is not specified, all files will be considered for rewriting |

Reviewer: ```suggestion
| `max-files-to-rewrite` | null | This option sets an upper limit on the number of eligible files that will be rewritten. If this option is not specified, all eligible files will be rewritten. |
```

Lastly, the [register_table](#register_table) procedure can be used to register the copied table in the target location with a catalog.

!!! warning

>     Iceberg tables with partition statistics files are not currently supported for path rewrite.

Reviewer 2: Unnecessary newline change?

Author: Yeah seems like my IDE is adding these

Author: Retried the same approach through online git file edit. from UI as well.  It seems like adding the new option is automatically showing line 1059 as a change.

Reviewer: There is an extra " " at the end of this line that you are removing.

"for path rewrite " vs "for path rewrite"

Reviewer: Double never mind, there it is
![image](https://github.com/user-attachments/assets/3525bf17-5d5b-45d4-a96e-24557b369990)

Author: @RussellSpitzer  I tried editing the line from Github repo , sublime etc but for some reason my commits wouldnt work :|
